app/controllers/admin/schedules_controller.rb

```diff
@@ -31,7 +31,16 @@ def index
 
     def collection
       return Schedule.where("1=0") unless json_request?
-      permissions.visible_schedules
+      if params[:enterprise_id]
+        filter_schedules_by_enterprise_id(permissions.visible_schedules, params[:enterprise_id])
+      else
+        permissions.visible_schedules
+      end
+    end
+
+    # Filter schedules by OCs with a given coordinator id
+    def filter_schedules_by_enterprise_id(schedules, enterprise_id)
+      schedules.select { |schedule| schedule.coordinators.map(&:id).include? enterprise_id.to_i }
```

Comment:
Is there anything stopping us from fetching this from DB? It will be orders of magnitude faster

Reply:
I opted to do this because I'd like to keep the permissions code in visible_schedules as is and not mix permissions with data fetching here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/e15ccdbd74a7e2582d0cdd251ac118567db95224/lib/open_food_network/permissions.rb#L135

Reply:
otherwise I think I'd have to mix the two parts in one query.

Reply:
Have you played with AR's `#merge`?

Reply:
ok, I'll give it try.